CodeGeneration/Sources/SyntaxSupport/ExprNodes.swift

```diff
@@ -1683,37 +1683,51 @@ public let EXPR_NODES: [Node] = [
 
   // switch-expr -> identifier? ':'? 'switch' expr '{'
   //   switch-case-list '}' ';'?
-  //
-  // This node represents both a 'switch' expression, as well as a 'switch'
-  // statement when wrapped in an ExpressionStmt node.
   Node(
     kind: .switchExpr,
     base: .expr,
     nameForDiagnostics: "'switch' statement",
+    documentation: """
+      A `switch` expression.
+
+      This node represents both a 'switch' expression.
+
+      ```swift
+      switch self.foo {
+      }
+      ```
```

Comment:
In https://github.com/apple/swift-syntax/pull/2179, I’m introducing an `### Examples` section to a few nodes, which would be a good fit here as well, I think.